File: content/problems/0454-4sum-ii/solutions/pair-sum-frequency/Solution.java
class Solution {
    public int fourSumCount(int[] nums1,int[] nums2,int[] nums3,int[] nums4){
        Map<Integer,Integer> counts=new HashMap<>();
        for(int a:nums1)for(int b:nums2)counts.put(a+b,counts.getOrDefault(a+b,0)+1);
        int answer=0;
        for(int c:nums3)for(int d:nums4)answer+=counts.getOrDefault(-c-d,0);
        return answer;
    }
}

File: content/problems/0454-4sum-ii/solutions/pair-sum-frequency/solution.cpp
class Solution {
    public: int fourSumCount(vector<int>& nums1,vector<int>& nums2,vector<int>& nums3,vector<int>& nums4){
        unordered_map<int,int> counts;
        for(int a:nums1)for(int b:nums2)++counts[a+b];
        int answer=0;
        for(int c:nums3)for(int d:nums4)answer+=counts[-c-d];
        return answer;
    }
};
